### PR TITLE
General slate changes (good to merge before cold flow)

### PR DESCRIPTION
**Pull request by mikh3x4**
Started adding descriptions and units to the slate.
I'll also add a tcp heartbeat and tcp newlines

Don't merge yet since its still a work in progress (unless you have better ideas)

### CODE REVIEW

**Comment by mikh3x4** [comment]
Added newilne tcp sperators. Goes along with [GUI branch](https://github.com/stanford-ssi/Olympus-GroundControl/tree/michal/newlines)

File: src/quail/config.h [hunk 1]
```diff
@@ -41,7 +41,7 @@ typedef enum {
     SMALL = 130,
     MEDIUM = 220, // Edelbrook
     LARGE = 180, // Pro BigShot
-    ABORT = 255 // don't pwm, just digital write
+    ABORT_VALVE = 255 // don't pwm, just digital write
```
Comment (Timvrakas) [inline src/quail/config.h:44]: this is why #defines are bad... should use enums

Comment (Timvrakas) [reply src/quail/config.h:44]: or these are enums.. hmm. I thought there was a way to make them not conflict.

File: src/quail/config.h [hunk 2]
```diff
@@ -60,12 +60,27 @@ typedef enum {
 
 // sequence
 #define UPDATE_SEQS 0b01
+#define MAWP 700
+#define FSM_PTM 100
+#define FSM_FREQ 10
+
+typedef enum {
```
Comment (Timvrakas) [inline src/quail/config.h:67]: They could be like: TANKSTATE_ABORT, TANKSTATE_IDLE, etc

File: src/quail/rx_tx_log/EthernetTask.cpp [hunk 3]
```diff
@@ -63,7 +63,8 @@ err_t EthernetTask::requestHandler(netconn *&conn, char *rcv, uint16_t len) {
         if (pckt.containsKey("cmd")) { // Command echo server
             cmdBuf.send(rcvCpy, MAX_CMD_LENGTH);
             // Serial.println(rcvCpy);
-            return netconn_write(conn, rcvCpy, len, NETCONN_COPY);
+            netconn_write(conn, rcvCpy, len, NETCONN_COPY);
+            return netconn_write(conn, "\n", 1, NETCONN_COPY);
```
Comment (Timvrakas) [inline src/quail/rx_tx_log/EthernetTask.cpp:67]: lol this is probably slow. Once I fix the ethernet GMAC bug I will try to improve.

File: src/quail/sensors/ThermalSensor.cpp [hunk 4]
```diff
@@ -3,7 +3,9 @@
 uint8_t ThermalSensor::cfg = UNCONFIGURED; // this sensor type is initially unconfigured
 
 ThermalSensor::ThermalSensor(Ad7124::InputSel ainp, SensorSlate &slate) : 
-    Sensor(slate.id.c_str(), ainp, Ad7124::AIN0Input, slate){};
+    Sensor(slate.id.c_str(), ainp, Ad7124::AIN0Input, slate){
+        slate.updateUnits("C");
```
Comment (Timvrakas) [inline src/quail/sensors/ThermalSensor.cpp:7]: cries in degrees-not-a-unit

File: src/quail/slate/SlateEndpoints.hpp [hunk 5]
```diff
@@ -9,10 +9,10 @@
 template <typename T>
 class EndPoint : public SlateKey<T>{
     public:
-        EndPoint(const std::string id, const std::string quailID, T init, bool editable) : 
-        SlateKey<T>(id, init), quailID(quailID), editable(editable) {}
+        EndPoint(const std::string id, const std::string quailID, T init, bool editable, const std::string desc = "none") :
```
Comment (Timvrakas) [inline src/quail/slate/SlateEndpoints.hpp:12]: I am vaugely concerned about how memory for this is being allocated, but who knows I guess.

**Review by Timvrakas — APPROVED** [review]
Yeah full send we can always fix problems later

**Pull request merged**